invenio_rdm_records/services/services.py

```diff
@@ -405,6 +405,32 @@ def reserve_pid(self, id_, identity, pid_type, pid_client=None):
             links_tpl=self.links_item_tpl,
         )
 
+    def resolve_pid(
+        self, id_, identity, pid_type, pid_prefix=None, pid_client=None
+    ):
+        """Resolve PID to a record."""
+        # PIDS-FIXME: should be migrated to self.record_cls.pids.resolve()
+        provider = self.get_provider(pid_type, client_name=pid_client)
+
+        pid_value = id_
+        if pid_prefix:
+            pid_value = f"{pid_prefix}/{pid_value}"
```

Comment:
Suggestion: move this up to the view. Like this, the service can stay generic and not having to know if a prefix exists or not.
You can also then put in common the call to the service in the resource and refactoring to 2 views to all the same function.